src/codegate/pipeline/sensitive_data/manager.py

```diff
@@ -0,0 +1,59 @@
+import json
+from typing import Dict, Optional
+import structlog
+from codegate.pipeline.sensitive_data.session_store import SessionStore
+
+logger = structlog.get_logger("codegate")
+
+
+class SensitiveData:
+    """Represents sensitive data with additional metadata."""
+
+    def __init__(self, original: str, service: Optional[str], type: Optional[str]):
+        self.original = original
+        self.service = service
+        self.type = type
+
+    def to_json(self) -> str:
+        """Serializes the object to a JSON string."""
+        return json.dumps({key: value for key, value in vars(self).items() if value is not None})
+
+    @staticmethod
+    def from_json(data: str) -> "SensitiveData":
+        """Deserializes from a JSON string to a SensitiveData object."""
+        obj = json.loads(data)
+        return SensitiveData(obj["original"], obj.get("service"), obj.get("type"))
```

Comment:
*nit:* I'm not a Python expert, but I believe we should be able to get this "for free" by extending `pydantic.BaseModel`.